File: USACO/Training/1.2/ride.py
"""
ID: piraadw1
LANG: PYTHON3
TASK: ride
"""
fin = open ('ride.in', 'r')
fout = open ('ride.out', 'w')

s1 = fin.readline()[:-1]

s2 = fin.readline()[:-1]

r1 = 1
for ch in s1:
    r1 *= ord(ch)-64

r2 = 1
for ch in s2:
    r2 *= ord(ch)-64

if r1%47==r2%47:
    s = "GO\n"
else:
    s = "STAY\n"

fout.write(s)
fout.close()

File: USACO/Training/1.2/ride.cpp
/*
ID: piraadw1
TASK: ride
LANG: C++                 
*/

#include <bits/stdc++.h>

using namespace std;

int main()
{
	ofstream fout ("ride.out");
    ifstream fin ("ride.in");
    string a, b;
    int out1=1, out2=1;
    fin >> a >> b;
    for (int i=0; i<a.size(); ++i)
    	out1*=a[i]-'A'+1;
   	for (int i=0; i<b.size(); ++i)
   		out2*=b[i]-'A'+1;
   	if (out1%47==out2%47)
   		fout << "GO" << endl;
   	else
   		fout << "STAY" << endl;
    return 0;
}